sdk/basekey_test.go

```diff
@@ -0,0 +1,334 @@
+	// Create wellknown configuration without base key
+	wellknownConfig := map[string]interface{}{
+		"some_other_config": "value",
+	}
+
+	mockService := newMockWellKnownService(wellknownConfig, nil)
+	s.sdk.wellknownConfiguration = mockService
+
+	// Call getBaseKey
+	baseKey, err := getBaseKey(s.T().Context(), s.sdk)
+
+	// Validate result
+	s.Require().True(mockService.called)
+	s.Require().Error(err)
+	s.Require().Nil(baseKey)
+	s.Require().Contains(err.Error(), errBaseKeyNotFound.Error())
+}
+
+func (s *BaseKeyTestSuite) TestGetBaseKeyInvalidBaseKeyFormat() {
+	wellknownConfig := map[string]interface{}{
+		baseKeyWellKnown: "invalid-base-key",
+	}
+
+	mockService := newMockWellKnownService(wellknownConfig, nil)
+	s.sdk.wellknownConfiguration = mockService
+
+	// Call getBaseKey
+	baseKey, err := getBaseKey(s.T().Context(), s.sdk)
+
+	// Validate result
+	s.Require().True(mockService.called)
+	s.Require().Error(err)
+	s.Require().Nil(baseKey)
+	s.Require().ErrorContains(err, errBaseKeyInvalidFormat.Error())
+}
+
+func (s *BaseKeyTestSuite) TestGetBaseKeyEmptyBaseKey() {
+	// Create empty base key map
+	wellknownConfig := map[string]interface{}{
+		baseKeyWellKnown: map[string]interface{}{},
+	}
+
+	mockService := newMockWellKnownService(wellknownConfig, nil)
+	s.sdk.wellknownConfiguration = mockService
+
+	// Call getBaseKey
+	baseKey, err := getBaseKey(s.T().Context(), s.sdk)
+
+	// Validate result
+	s.Require().True(mockService.called)
+	s.Require().Error(err)
+	s.Require().Nil(baseKey)
+	s.Require().ErrorContains(err, errBaseKeyEmpty.Error())
+}
+
+func (s *BaseKeyTestSuite) TestGetBaseKeyMissingPublicKey() {
+	// Create base key without public_key field
+	wellknownConfig := map[string]interface{}{
+		baseKeyWellKnown: map[string]interface{}{
+			"kas_uri": "https://test-kas.example.com",
+			// Missing public_key field
+		},
+	}
+
+	mockService := newMockWellKnownService(wellknownConfig, nil)
+	s.sdk.wellknownConfiguration = mockService
+
+	// Call getBaseKey
+	baseKey, err := getBaseKey(s.T().Context(), s.sdk)
+
+	// Validate result
+	s.Require().True(mockService.called)
+	s.Require().Error(err)
+	s.Require().Nil(baseKey)
+	s.Require().ErrorContains(err, errBaseKeyInvalidFormat.Error())
+}
+
+func (s *BaseKeyTestSuite) TestGetBaseKeyInvalidPublicKey() {
+	// Create base key with invalid public_key (string instead of map)
+	wellknownConfig := map[string]interface{}{
+		baseKeyWellKnown: map[string]interface{}{
+			"kas_uri":        "https://test-kas.example.com",
+			baseKeyPublicKey: "invalid-public-key", // Should be a map
+		},
+	}
+
+	mockService := newMockWellKnownService(wellknownConfig, nil)
+	s.sdk.wellknownConfiguration = mockService
+
+	// Call getBaseKey
+	baseKey, err := getBaseKey(s.T().Context(), s.sdk)
+
+	// Validate result
+	s.Require().True(mockService.called)
+	s.Require().Error(err)
+	s.Require().Nil(baseKey)
+	s.Require().ErrorContains(err, errBaseKeyInvalidFormat.Error())
```

Comment:
Can we please update all these error assertions to `ErrorIs(err, errBaseKeyInvalidFormat)` instead of `contains`?